tests/integration/profiling/rule_based_profilers/test_run_profiler_notebook.py

```diff
@@ -10,6 +11,30 @@
 logger = logging.getLogger(__name__)
 
 
+def clean_up_test_files(
+    base_dir: str, additional_files_to_clean_up: Optional[List[str]] = None
+) -> None:
```

Comment:
Could we possibly make this a bit more generic? Rather than defining specific files and directories here, we rely on the arguments to determine what needs to be removed?

```python
def clean_up_test_files(paths: List[str]) -> None:
    for path in paths:
        if not os.path.exists(path):
            continue
        if os.path.isdir(path):
            # remove dir
        else:
            # remove file
```

Definitely a personal opinion but I like explicitly seeing which files are going to be cleaned up in that `finally` block. Perhaps a bit of code duplication is introduced but I think it becomes abundantly clear that we're cleaning up X, Y, and Z file and have no concern about side effects.